### DISCUSSION TITLE
How do you specify a locale when debugging

### DISCUSSION THREAD

**Discussion by markdemich** [BowlerHatLLC/vscode-as3mxml#680] [Q&A]
My app supports multiple locales.  Is there anyway to specify the locale when running from the ide?

**Comment by joshtynjala** [comment]
- Are you using a framework like Flex?
- How do you detect the locale normally when not running from the IDE?

**Reply by markdemich** [reply]
It'a a flex app that was in a browser that I'm wrapping with Air now. We are passing it at runtime while "invoking" from a web uri.  I'm looking for a way to set it while debugging in vscode.

**Reply by joshtynjala** [reply]
You can set the `args` field in _launch.json_ for your "swf" debug configuration to pass command line arguments to your AIR app.

https://github.com/BowlerHatLLC/vscode-as3mxml/wiki/launch.json-fields-for-SWF-debugging#args

Your AIR app can listen for `InvokeEvent.INVOKE` from `NativeApplication.nativeApplication` to access its command line arguments.

**Reply by markdemich** [reply]
Thanks.  Actually, I may be getting ahead of myself.  I have yet to figure out how to bundle the resources.  I'm converting a project that used to be set up on FlashBuilder. I had multiple projects what were compiled into individual SWCs because I shared them.  I then had a project for my Air application.  When moving to vscode with the as3mxl extension, I had trouble getting that setup to work without jumping through hoops to compile each module everytime I made a change.  I hadn't yet gotten to the point of testing my localization for which I have many languages.  So to just make things simpler I had aborted on making the individual modules, but instead just added those projects src folders to my source-path.  I got things running I had to point directly to each en_US locale folder.  When I tried to use xxx/locale/{locale} like I did in Flashbuilder it was not accepted.  So just for the heck of, I copied all the bundles into one locale folder and included that locale folder.  When I tried to compile, the [ResourceBundle("xxxx")] in all my sub projects gave an error, "Error unable to resolve resource bundle "xyz""  

I'm probably doing something wrong so I'm curious how are people dealing with localization.  It was pretty easy with flashbuilder, but I'm struggling here to get things set up.

**Reply by joshtynjala** [reply]
Yes, I see that when vscode-as3mxml tries to copy source path assets to the output directory, it doesn't know how to handle the `{locale}` token in the source-path and says that the source path is invalid. I think I should at least change it to skip that path and continue with a warning. Ideally, it would resolve `{locale}`, but not failing would be a good first step.

**Reply by joshtynjala** [reply]
The `{locale}` issue is fixed in commit fec9956ee6fd76fc0aef19228b20931f6b6be21b. It will be included whenever I release the next update.

In the meantime, you could try downloading and installing a nightly build from Github Actions.

**Reply by markdemich** [reply]
Thanks for the quick fix.  I also found a work around. I was able to use additionalOptions to add this to the source path.

I wound up just adding this to my additionalOptions string.

--source-path+=locale/{locale}

**Comment by markdemich** [comment]
Thanks. I'll try first thing in the morning.
